spec/rubocop/cop/style/stderr_puts_spec.rb

```diff
@@ -23,4 +23,23 @@
       $stderr.puts
     RUBY
   end
+
+  it "registers an offense when using `STDERR.puts('hello')`" do
+    expect_offense(<<-RUBY.strip_indent)
+      STDERR.puts('hello')
+      ^^^^^^^^^^^ Use `warn` instead of `$stderr.puts` or `STDERR.puts` to allow such output to be disabled.
+    RUBY
+  end
+
+  it 'autocorrects `STDERR.puts` to `warn`' do
+    new_source = autocorrect_source("STDERR.puts('hello')")
+
+    expect(new_source).to eq "warn('hello')"
+  end
+
+  it 'registers no offense when using `STDERR.puts` with no arguments' do
+    expect_no_offenses(<<-RUBY.strip_indent)
+      STDERR.puts
+    RUBY
+  end
```

Comment:
I believe that `STDERR.puts` with no arguments *should* register an offense. Printing a newline should be no different that printing anything else. Perhaps this should be discussed in a separate PR, though?

Reply:
Agreed on both counts. I can't think of a situation where it would be useful to allow `STDERR.puts` without arguments but prohibit it with arguments. But for this PR, I think it's best to follow the example of the legacy logic.